src/Raven.Client/Documents/Operations/MaintenanceOperationExecutor.cs

```diff
@@ -75,7 +75,7 @@ public async Task<Operation> SendAsync(IMaintenanceOperation<OperationIdResult>
                 var command = operation.GetCommand(RequestExecutor.Conventions, context);
 
                 await RequestExecutor.ExecuteAsync(command, context, sessionInfo: null, token: token).ConfigureAwait(false);
-                return new Operation(RequestExecutor, () => _store.Changes(_databaseName), RequestExecutor.Conventions, command.Result.OperationId);
+                return new Operation(RequestExecutor, () => _store.Changes(_databaseName), RequestExecutor.Conventions, command.Result.OperationId, command.SelectedNodeTag ?? command.Result.OperationNodeTag);
```

Comment:
Shouldn't this be the other way around? First take the node tag from result, then from command?

Reply:
If we select node to execute => execute on this (selected) node
if we dont select => execute on command.Result.OperationNodeTag